sympy/tensor/indexed.py

```diff
@@ -364,12 +366,19 @@ class IndexedBase(Expr, NotIterable):
     is_Atom = True
 
     def __new__(cls, label, shape=None, **kw_args):
+        from sympy import MatrixBase, NDimArray
+
         if isinstance(label, string_types):
             label = Symbol(label)
         elif isinstance(label, Symbol):
             pass
+        elif isinstance(label, (MatrixBase, NDimArray)):
+            return label
+        elif isinstance(label, collections.Iterable):
+            from sympy import ImmutableDenseNDimArray
+            return ImmutableDenseNDimArray(label)
         else:
-            label = _sympify(label)
+            label = sympify(label)
```

Comment:
@asmeurer: That's not obvious from the name alone.  A name such as `sympify_without_string_conversion` (or maybe `sympify_no_string_conv`) may be better, rather than `_sympify`.